flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/util/ThriftObjectConversionsTest.java

```diff
@@ -34,4 +107,249 @@ public void testConvertSessionHandle() {
+                DataTypeSpec.newSpec()
+                        .forType(TINYINT())
+                        .forValue((byte) 3)
+                        // TINYINT is the alias of the BYTE in Hive.
+                        .expectSqlType(Types.BINARY),
+                DataTypeSpec.newSpec()
+                        .forType(SMALLINT())
+                        .forValue((short) 255)
+                        .expectSqlType(Types.SMALLINT),
+                DataTypeSpec.newSpec().forType(INT()).forValue(1994).expectSqlType(Types.INTEGER),
+                DataTypeSpec.newSpec()
+                        .forType(BIGINT())
+                        .forValue(13214991L)
+                        .expectSqlType(Types.BIGINT),
+                DataTypeSpec.newSpec()
+                        .forType(FLOAT())
+                        .forValue(1024.0f)
+                        .expectSqlType(Types.FLOAT)
+                        .expectValue(1024.0),
+                DataTypeSpec.newSpec()
+                        .forType(DOUBLE())
+                        .forValue(2048.1024)
+                        .expectSqlType(Types.DOUBLE),
+                DataTypeSpec.newSpec()
+                        .forType(DECIMAL(9, 6))
+                        .forValue(DecimalData.fromBigDecimal(new BigDecimal("123.456789"), 9, 6))
+                        .expectSqlType(Types.DECIMAL)
+                        .expectValue("123.456789"),
+                DataTypeSpec.newSpec()
+                        .forType(STRING())
+                        .forValue(StringData.fromString("Hello World"))
+                        .expectSqlType(Types.VARCHAR)
+                        .expectValue("Hello World"),
+                DataTypeSpec.newSpec()
+                        .forType(BYTES())
+                        .forValue("Flink SQL Gateway".getBytes(StandardCharsets.UTF_8))
+                        .expectSqlType(Types.BINARY)
+                        .expectValue(
+                                new String("Flink SQL Gateway".getBytes(StandardCharsets.UTF_8)),
+                                "Flink SQL Gateway".getBytes(StandardCharsets.UTF_8)),
+                DataTypeSpec.newSpec()
+                        .forType(DATE())
+                        .forValue((int) LocalDate.parse("2022-02-22").toEpochDay())
+                        .expectSqlType(Types.DATE)
+                        .expectValue("2022-02-22"),
+                DataTypeSpec.newSpec()
+                        .forType(TIMESTAMP(4))
+                        .forValue(
+                                TimestampData.fromLocalDateTime(
+                                        LocalDateTime.parse("2020-05-11T12:00:12.1234")))
+                        .expectSqlType(Types.TIMESTAMP)
+                        .expectValue("2020-05-11 12:00:12.1234"),
+                DataTypeSpec.newSpec()
+                        .forType(DataTypes.MAP(DataTypes.STRING(), DataTypes.STRING()))
```

Comment:
Could you create an issue to support non-string keys in the future?

Reply:
I open a ticket in the [FLINK-28723](https://issues.apache.org/jira/browse/FLINK-28723).